src/server/func_services/func_server.js

```diff
@@ -417,21 +417,24 @@ function _make_rpc_options(req) {
 function _make_aws_config(req) {
     // TODO: need to change when we start to run funcs outside the
     // cluster (should use a proper hint based on agent location)
-    const ep_host = addr_utils.get_base_address(req.system.system_address).hostname;
-    const ep_port = parseInt(process.env.ENDPOINT_PORT, 10) || 80;
+    const ep_addr = addr_utils.get_base_address(req.system.system_address, {
+        hint: 'INTERNAL',
+        service: 's3',
+        api: 's3',
+        protocol: 'https',
+    });
     const account = system_store.data.get_by_id(req.func.exec_account);
     const account_keys = account.access_keys[0];
     return {
         region: 'us-east-1',
-        endpoint: `http://${ep_host}:${ep_port}`,
+        endpoint: ep_addr.toString(),
```

Comment:
Why toString()? what is the type of ep_addr?

Reply:
ep_addr is URL

Reply:
ah ok, so I would suggest using href instead to get the url string because it is more explicit.
see https://nodejs.org/docs/latest-v10.x/api/url.html#url_url_href
```suggestion
        endpoint: ep_addr.href,
```

Reply:
👍